diff --git a/resources/lib/modules/system.py b/resources/lib/modules/system.py
index a4ed980..d2f00df 100755
--- a/resources/lib/modules/system.py
+++ b/resources/lib/modules/system.py
@@ -77,7 +77,7 @@ class system:
                     'settings': {'hostname': {
                         'order': 1,
                         'name': 32190,
-                        'value': 'OpenELEC',
+                        'value': 'TLBB',
                         'action': 'set_hostname',
                         'type': 'text',
                         'validate': '^([a-zA-Z0-9](?:[a-zA-Z0-9-\.]*[a-zA-Z0-9]))$',
