diff --git a/service.py b/service.py
index 9590718..63e0e21 100755
--- a/service.py
+++ b/service.py
@@ -92,8 +92,8 @@ class service_thread(threading.Thread):
 
             self.oe.dbg_log('_service_::run', 'enter_function', 0)
             
-            if self.oe.read_setting('openelec', 'wizard_completed') == None:
-                threading.Thread(target=self.oe.openWizard).start()
+ #           if self.oe.read_setting('openelec', 'wizard_completed') == None:
+ #               threading.Thread(target=self.oe.openWizard).start()
         
             while self.stopped == False:
 
