diff --git a/Makefile.am b/Makefile.am
index ac7924e..012ee8a 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -3026,7 +3026,6 @@ libsystemd_network_la_SOURCES = \
 	src/systemd/sd-icmp6-nd.h \
 	src/systemd/sd-dhcp6-client.h \
 	src/systemd/sd-dhcp6-lease.h \
-	src/systemd/sd-pppoe.h \
 	src/libsystemd-network/sd-dhcp-client.c \
 	src/libsystemd-network/sd-dhcp-server.c \
 	src/libsystemd-network/dhcp-network.c \
@@ -3041,7 +3040,6 @@ libsystemd_network_la_SOURCES = \
 	src/libsystemd-network/ipv4ll-network.c \
 	src/libsystemd-network/ipv4ll-packet.c \
 	src/libsystemd-network/ipv4ll-internal.h \
-	src/libsystemd-network/sd-pppoe.c \
 	src/libsystemd-network/network-internal.c \
 	src/libsystemd-network/network-internal.h \
 	src/libsystemd-network/sd-icmp6-nd.c \
