diff --git a/Makefile b/Makefile
index 7045a2e..f6df74a 100644
--- a/Makefile
+++ b/Makefile
@@ -71,7 +71,7 @@ $(BUILDDIR)/$(ADDON_NAME)/resources/skins: $(BUILDDIR)/$(ADDON_NAME)/resources/s
 
 $(BUILDDIR)/$(ADDON_NAME)/resources/skins/Default/media/default:
 	mkdir -p $(BUILDDIR)/$(ADDON_NAME)/resources/skins/Default/media/default
-	cp textures/$(DISTRONAME)/*.png $(BUILDDIR)/$(ADDON_NAME)/resources/skins/Default/media/default
+	cp textures/OpenELEC/*.png $(BUILDDIR)/$(ADDON_NAME)/resources/skins/Default/media/default
 
 $(BUILDDIR)/$(ADDON_NAME)/resources/skins/Default/media/icons:
 	mkdir -p $(BUILDDIR)/$(ADDON_NAME)/resources/skins/Default/media/icons
